mantidproject/mantid
# provide a user file column in the data table

Description of work.

The isis v2 gui currently does not allow different user files to be used on each row. A new column should be added allowing this to be specified on a row by row basis. If no user files is specified the default should be used. If a user file is specified it should be used unmodified.
**To test:**

- get the test data from \\olympic\Babylon5\Scratch\Matthew Andrew\Row_user_test
- in mantid go to Interfaces->Isis v2 experimental
- select `batch_file` as the batch file.
- select `USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT_gravON as the user file`
- In the first row enter `USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT_gravOFF.txt` in the userfile column
- Go to Settings->Q and check that the box for gravity is checked
- Go to Settings->State Diagnostics and click Update row selection
- Select row 0 and check that convert_to_q.use_gravity is false
- Select row 1 and check that convert_to_q.use_gravity is true


Fixes #20547. 

**Release Notes** 


---

#### Reviewer ####
Release notes to be added once 
Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
